source/margarita/upgrading.rst

```diff
@@ -0,0 +1,467 @@
+
+      mv conf/pillar/staging/secrets.sls staging-secrets.sls
+
+#. Rename ``env.sls``:
+
+   .. code-block:: bash
+
+      mv conf/pillar/staging/env.sls conf/pillar/staging.sls
+      rmdir conf/pillar/staging
+
+#. Update the ``conf/pillar/top.sls`` file:
+
+   .. code-block:: diff
+
+      Modified   conf/pillar/top.sls
+      diff --git a/conf/pillar/top.sls b/conf/pillar/top.sls
+      index 720b942..0db9e8a 100644
+      --- a/conf/pillar/top.sls
+      +++ b/conf/pillar/top.sls
+      @@ -10,4 +10,3 @@ base:
+           - match: grain
+      -    - staging.env
+      -    - staging.secrets
+      +    - staging
+         'environment:production':
+
+#. Commit and redeploy:
+
+   .. code-block:: bash
+
+      fab staging deploy
+
+Miscellaneous work
+------------------
+
+At this point you hopefully have a working server, but there is still some work to do.
+
+1. Port any useful functions in ``fabfile.py.old`` to the new fabfile, then remove the old one.
+
+#. Get a copy of the ``Makefile`` from the project template, porting any functions in your existing
+   one to the new one, if needed.
+
+#. Review everything in ``salt.old`` to see which pieces are specific to your project and need to
+   be added back into salt. If any of it is generally useful (i.e. setting up a service that
+   might be used on another project), then consider adding a PR to margarita so this config can
+   be completely removed from your project.
+
+   This part is difficult to generalize... Sorry. You kinda have to look in each state file and
+   make sure that service is properly accounted for in the new Margarita system.
+
+#. Look at the following files in django-project-template to see if your project could benefit
+   from any changes:
+
+   * .coveragerc
+   * .gitignore
+   * README.rst
+   * setup.cfg
+   * .travis.yml (look at project.travis.yml)
+
+Updating Production
+-------------------
+
+If staging updates successfully, these are a few steps you'll need to get production updated.
+
+1. Upgrade salt. Follow the steps in the :ref:`Upgrade Salt <upgrade-salt>` pathway above, but on
+   the production machine.
+
+#. Encrypt secrets: Follow the steps in the :ref:`Encrypt Secrets <encrypt-secrets>` pathway above.
+   Be sure to keep a copy of the unencrypted production secrets until things are working.
+
+#. Commit those changes and then run::
+
+     fab production sync
+     fab production deploy
+
+
+Troubleshooting
+---------------
+
+Here are some issues that may or may not come up during your upgrade. As we find new issues, we
+should update the docs above (if they are general), or add them here, if they aren't general or
+we're not sure.
+
+* ``newrelic_license_key`` must be capitalized. Some projects have a secret for
+  ``newrelic_license_key``, but the current margarita uses ``NEW_RELIC_LICENSE_KEY``
+* NewRelic settings may need adjusting, which you can do via environment variables (see the
+  margarita README for details)
```

Comment:
I just opened an issue (https://github.com/caktus/developer-documentation/issues/21) to move that documentation from the Margarita README to the development docs where the other information about setting up to use Margarita states lives.